src/dev/run_extract_default_translations.js

```diff
@@ -0,0 +1,343 @@
+import { jsdom } from 'jsdom';
+import { parse } from '@babel/parser';
+import traverse from 'babel-traverse';
+import * as babelTypes from 'babel-types';
+import { promisify } from 'util';
+
+import { run } from './run';
+
+const defaultMessageKey = 'defaultMessage';
+const angularExpressionRegEx = /(\{)\{+([\s\S]*?)(\})\}+/g;
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+const makeDir = promisify(fs.mkdir);
+const globProm = promisify(glob);
+
+run(async () => {
+  for (const inputPath of process.argv.slice(2)) {
+    await extractDefaultTranslations(inputPath);
+  }
+});
+
+async function isPathExists(path) {
+  await promisify(fs.access)(path, fs.constants.F_OK);
+}
+
+function addMessageToMap(targetMap, key, value) {
+  if (targetMap.has(key) && targetMap.get(key) !== value) {
+    throw new Error(`Default messages are different for id "${key}"`);
+  }
+  targetMap.set(key, value);
+}
+
+function parseConditionalOperatorAST(subTree) {
+  const ids = [];
+
+  if (babelTypes.isStringLiteral(subTree)) {
+    ids.push(subTree.value);
+  } else if (babelTypes.isConditionalExpression(subTree)) {
+    ids.push(...parseConditionalOperatorAST(subTree.consequent));
+    ids.push(...parseConditionalOperatorAST(subTree.alternate));
+  }
+
+  return ids;
+}
+
+function parseConditionalOperatorExpression(expression) {
+  let ids = [];
+
+  traverse(parse(expression), {
+    enter(path) {
+      if (babelTypes.isDirectiveLiteral(path.node)) {
+        ids = [path.node.value];
+        path.stop();
+      } else if (babelTypes.isConditionalExpression(path.node)) {
+        ids = parseConditionalOperatorAST(path.node);
+        path.stop();
+      }
+    },
+  });
+
+  return ids;
+}
+
+function parseFilterObjectExpression(expression) {
+  let defaultMessage = '';
+
+  const filterObjectAST = parse(`+${expression}`);
+  traverse(filterObjectAST, {
+    enter(path) {
+      if (babelTypes.isObjectExpression(path.node)) {
+        path.node.properties.forEach(property => {
+          if (isPropertyWithKey(property, defaultMessageKey) && babelTypes.isStringLiteral(property.value)) {
+            defaultMessage = property.value.value;
+          }
+        });
+      }
+    }
+  });
+
+  return defaultMessage;
+}
+
+function getFilterMessages(htmlContent) {
+  const expressions = (htmlContent.match(angularExpressionRegEx) || [])
+    .filter(expression => expression.indexOf('|') !== -1)
+    .filter(expression => expression.indexOf('i18n') !== -1)
+    .map(expression => expression.slice(2, -2).trim());
+
+  const defaultMessagesMap = new Map();
+
+  expressions.forEach(expression => {
+    const [idsExpression] = expression.match(/[\s\S]+(?=\|[\s]*i18n)/g) || [];
+    const [filterObjectExpression] = expression.match(/(?<=\|[\s]*i18n[\s]*:)[\s\S]+/g) || [];
+
+    if (!filterObjectExpression || !idsExpression) {
+      throw new Error('Angular.js expression syntax error');
```

Comment:
__nit:__ please add a little bit more details to the error so that it's easier to guess what the problem is.